modules/nw-configuring-qinq-sriov-proc.adoc

```diff
@@ -0,0 +1,196 @@
+* You have installed the SR-IOV Network Operator.
+
+.Procedure
+
+. Create an `SriovNetworkNodePolicy` object, and then save the YAML in the `<name>-sriov-node-network.yaml` file. Replace `<name>` with the name for this configuration.
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: sriovnetpolicy-810
+  namespace: openshift-sriov-network-operator
+spec:
+  deviceType: netdevice
+  nicSelector:
+    pfNames:
+      - ens5f0#0-9
+  nodeSelector:
+    node-role.kubernetes.io/worker-cnf: ""
+  numVfs: 10
+  priority: 99
+  resourceName: resource810
+----
+
+. Open a seperate terminal window and monitor the sync status of the SR-IOV network node state for the specified node in the `openshift-sriov-network-operator` namespace by running the following command: 
++
+[source,terminal]
+----
+$ watch -n 1 'oc get sriovnetworknodestates -n openshift-sriov-network-operator <node_name> -o jsonpath="{.status.syncStatus}"'
+----
+
+. Create the `SriovNetworkNodePolicy` object:
++
+[source,terminal]
+----
+$ oc create -f <name>-sriov-node-network.yaml
+----
++
+After applying the sync status indicates a change from `InProgress to: Succeeded`.
+
+. Create a `SriovNetwork` object, setting the outer VLAN referred to as the S-tag, or `Service Tag`, as it belongs to the infrastructure.
++
+[NOTE]
+====
+You need to configure this VLAN on the switch's trunk interface. Additionally, some switches may need further configuration to support QinQ tagging.
+====
+
+.. Using the following YAML example,create a file named `nad-sriovnetwork-1ad-810.yaml`:
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetwork
+metadata:
+  name: sriovnetwork-1ad-810
+  namespace: openshift-sriov-network-operator
+spec:
+  ipam: '{}'
+  vlan: 171 <1>
+  vlanProto: "802.1ad" <2>
+  networkNamespace: default
+  resourceName: resource810
+----
++
+<1> This sets the S-tag VLAN tag to 171. 
+<2> Specifies the VLAN protocol to assign to the VF. Allowed values are `802.1ad` and `802.1q`. The default is `802.1q`. 
+
+.. To create the object, enter the following command:
++
+[source,terminal]
+----
+$ oc create -f nad-sriovnetwork-1ad-810.yaml
+----
+
+. Create a Network Attachment Definition with an inner VLAN. The inner VLAN is often referred to as the C-tag, or `Customer Tag`, as it belongs to the Network Function.
+
+.. Using the following YAML example, create a file named `nad-cvlan100.yaml`:
++
+[source,yaml]
+----
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: nad-cvlan100
+  namespace: default
+spec:
+  config: |
+    {
+      "cniVersion": "0.3.1",
+      "name": "vlan-100",
+      "plugins": [
+        {
+          "type": "vlan",
+          "master": "net1", <1>
+          "vlanId": 100,
+          "linkInContainer": true, 
+          "ipam": {"type": "static"}
+        }
+      ]
+    }
+----
++
```

Comment:
The nad I used for testing looks a bit different.

apiVersion: k8s.cni.cncf.io/v1
kind: NetworkAttachmentDefinition
metadata:
  name: nad-cvlan100
  namespace: cni-test
spec:
  config: '{
            "name": "vlan-100",
            "cniVersion": "0.3.1",
            "type": "vlan",
            "linkInContainer": true,
            "master": "net1",
            "vlanId": 100, 
            "ipam": {"type": "static"}
            }'

Reply:
@kquinn1204, should we use this one instead?

Reply:
I not sure there is a difference. I just dont like to use the config command in the nad.